### ISSUE TITLE
Group Transform Bug

### ISSUE THREAD

**Issue by FeaRCODE** [kangax/fabric.js#2964]
## Version
1.6.1

## Test Case
https://jsfiddle.net/6269qpdu/

## Steps to reproduce
Select (group) the two rectangle and randomly transform them (i.e. squashing the entire object as small as possible). **Note:** I had to do the process multiple times, sometimes fast for the bug to occur. When this happens i get something that looks like this:

![image](https://cloud.githubusercontent.com/assets/6901336/15210327/8552b554-17fb-11e6-90b1-9b94b2b2b5b2.png)

I think this has something to do with me changing the padding here: `fabric.Object.prototype.padding = 5;`. When I remove this, everything seems to work fine. This bring me to a better point. Is there a decent way other than this to override these values?

## Expected Behavior
The group should normally transform.

## Actual Behavior
When grabbing the top/left/right controls and squashing the group, the entire group disappears and jumps to the top left of the canvas.
![dit fiddle - jsfiddle20160512045126](https://cloud.githubusercontent.com/assets/6901336/15210720/5031ce26-17fd-11e6-929b-83c42f939e89.gif)

**Comment by asturur** [comment]
They way you override the padding is ok.
Can you make the bug to happen in reproducible way?

**Comment by asturur** [comment]
I found it.
Group them, skew from right handle going up,
Then from left handle flip over.

**Comment by FeaRCODE** [comment]
@asturur Great! I posted a horrible GIF showing at least what is happening (edited)

**Comment by FeaRCODE** [comment]
Also, I would like to note on my project, the bug is worse. All i have to do is attempt to flip anywhere and it occurs. I am not sure why the JSFIDDLE take a few tries to get it to bug out.

**Comment by asturur** [comment]
Something hard to find out. I was about to release 1.6.2 but i have to fix this first.

**Comment by FeaRCODE** [comment]
For some reason it goes away "I think" when you set the padding to 0. Also, I'm not sure if this is related, but now my backgroundImage for my canvas doesn't scale anymore with my canvas.

**Comment by asturur** [comment]
![image](https://cloud.githubusercontent.com/assets/1194048/15211226/60106d10-183a-11e6-8fb6-b6dc5126bb20.png)

This is killing it.
When you scale down to flip, you arrive at the point that you have twice the padding ( from object and group ), Object has scale nearly 0, but we are still trying to reduce it and something bad happen. A division by 0 probably

**Comment by asturur** [comment]
No ok, bug is generic. Remove padding, group object, do not transform, scale to flip, go slow, when you cross the border of 0 it happens. :( This evening Cinema, tomorrow morning i'll look at it.

**Comment by FeaRCODE** [comment]
@asturur did you find what was causing this issue?

**Comment by asturur** [comment]
no movie finished to late to wake up early before work and do it. i hope this weekend i can clear out and release.

**Comment by FeaRCODE** [comment]
That is fine, I also think I found a bug regarding cloning groups, however I posted a question on StackOverflow regarding it if you would like to glance at it.

**Comment by asturur** [comment]
i saw the question, i tried to understand what is going on. I put it in the list of things to verify.

**Comment by FeaRCODE** [comment]
I attached a jsfiddle to show what is going on. I also did not mean to close the issue :)

**Comment by asturur** [comment]
This is a very strange bug and i do not understand why it did not came out before.
The bug is active since we introduced skew, and this because we started dividing some factor by actual dimensions. When going to flip actual dimensions go near 0, and dividing by 0 create the NaN value and ruin the rendering.

The fact is that we have a parameter that saves us and this is minScaleLimit. Now for objects bigger than 250 pixels ( very common ) this parameter is rounded to 0.
`Math.round(1/250, 2) = 0`
